File: src/com/l2kt/gameserver/network/clientpackets/RequestHennaRemove.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.model.item.Henna;
import com.l2kt.gameserver.network.SystemMessageId;

/**
 * format cd
 */
public final class RequestHennaRemove extends L2GameClientPacket
{
	private int _symbolId;
	
	@Override
	protected void readImpl()
	{
		_symbolId = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player activeChar = getClient().getActiveChar();
		if (activeChar == null)
			return;
		
		for (int i = 1; i <= 3; i++)
		{
			Henna henna = activeChar.getHenna(i);
			if (henna != null && henna.getSymbolId() == _symbolId)
			{
				if (activeChar.getAdena() >= (henna.getPrice() / 5))
				{
					activeChar.removeHenna(i);
					break;
				}
				activeChar.sendPacket(SystemMessageId.YOU_NOT_ENOUGH_ADENA);
			}
		}
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestHennaRemove.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.network.SystemMessageId

/**
 * format cd
 */
class RequestHennaRemove : L2GameClientPacket() {
    private var _symbolId: Int = 0

    override fun readImpl() {
        _symbolId = readD()
    }

    override fun runImpl() {
        val activeChar = client.activeChar ?: return

        for (i in 1..3) {
            val henna = activeChar.getHenna(i)
            if (henna != null && henna.symbolId == _symbolId) {
                if (activeChar.adena >= henna.price / 5) {
                    activeChar.removeHenna(i)
                    break
                }
                activeChar.sendPacket(SystemMessageId.YOU_NOT_ENOUGH_ADENA)
            }
        }
    }
}